docs/stdout-exporter-design.md

```diff
@@ -0,0 +1,223 @@
+                        << "\n  duration      : " << span->GetDuration().count() 
+                        << "\n  description   : " << span->GetDescription() 
+                        << "\n  status        : " << span->GetStatus()
+                        << "\n  attributes    : " << span->GetAttributes() << "\n}"
+                        << "\n";
+            }
+            
+            if(time.now() - startTime > 30s) {
+                return sdktrace::ExportResut::kFailure;
+            }
+        }
+    
+        return sdktrace::ExportResult::kSuccess;
+    }
+
+    void Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept
+    {
+        isShutdown = true;
+    }
+
+};
+
+
+```
+
+## MetricsExporter
+
+The MetricsExporter has the same requirements as the SpanExporter. The exporter will go through the different metric instruments and print the value stored in their aggregators to stdout, for simplicity only Counter is shown here, but all aggregators will be implemented. Counter, Gauge, MinMaxSumCount, Sketch, Histogram and Exact Aggregators will be supported.
+
+Exports a batch of telemetry data. Protocol exporters that will implement this function are typically expected to serialize and transmit the data to the destination.
+
+![SDK Data Path](./images/DataPath.png)
+
+### `Export(batch of Records)`
+
+Export() will never be called concurrently for the same exporter instance. Export() can be called again only after the current call returns.
+
+Export() must not block indefinitely, there must be a reasonable upper limit after which the call must time out with an error result (`Failure`).  `exporterTimeoutMillis` - how long the export can run before it is cancelled. The default value is `30000`.
+
+The specification states: Any retry logic that is required by the exporter is the responsibility of the exporter. The default SDK SHOULD NOT implement retry logic, as the required logic is likely to depend heavily on the specific protocol and backend the spans are being sent to.
+
+The MetricsExporter is called through the Controller in the SDK data path. The exporter will either be called on a regular interval in the case of a push controller or through manual calls in the case of a pull controller.
+
+### `Shutdown()`
+
+Shuts down the exporter. Called when SDK is shut down. This is an opportunity for exporter to do any cleanup required.
+
+`Shutdown` should be called only once for each `Exporter` instance. After the call to `Shutdown` subsequent calls to `Export` are not allowed and should return a `Failure` result.
+
+`Shutdown` should not block indefinitely (e.g. if it attempts to flush the data and the destination is unavailable). Language library authors can decide if they want to make the shutdown timeout configurable.
+
+In the StdoutExporter there is no cleanup to be done, so there is no need to use the timeout within the `Shutdown` function as it will never be blocking.
```

Comment:
Depending on whether we need to _flush_ the console.

Reply:
We would only need to flush if there are asynchronous calls to export() right?

Reply:
When working with streams should we flush the stream when we shutdown the exporter? Or should we leave this cleanup to the user since they also have the stream? @maxgolov